components/assetstore-controller-manager/pkg/loader/package_loader.go

```diff
@@ -0,0 +1,182 @@
+package loader
```

Comment:
Does it make sense to move the logic to the `loader.go` file? I know the file will be bigger, but... I like when all methods used are in single file.

Reply:
Okay, so maybe we could transform the `loader` to be composed of `baseLoader`, `pkgLoader` and `singleLoader`. It would be cleaner to read it